java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -757,6 +760,79 @@ public static Table concatenate(Table... tables) {
+   *   inputTable  =   { 10, 20, 20, 30, 50 }
+   *   valuesTable =   { 20 }
+   *   result      =   { 1 }
+   *
+   *  Multi Column:
+   *      idx                0    1    2    3    4
+   *   inputTable      = {{  10,  20,  20,  20,  20 },
+   *                      { 5.0,  .5,  .5,  .7,  .7 },
+   *                      {  90,  77,  78,  61,  61 }}
+   *   valuesTable     = {{ 20 },
+   *                      { .7 },
+   *                      { 61 }}
+   *   result          = {  3 }
+   * NaNs in column values do not work especially
+   * if the table has multiple columns.
+   * Strings and String categories do not work for this method. If the input table is
+   * unsorted the results are wrong. Types of columns can be of mixed data types and
+   * the result gives the index of the value pertaining to the matching type column
+   * in the input table.
+   * @param inputTable the sorted table as input for bounds
+   * @param valueTable the table of values that need to be inserted
+   * @param descFlags indicates the ordering of the column(s), true if descending
+   * @param nullsAsLargest true if nulls are assumed largest
+   * @return ColumnVector with lower bound indices for all rows in valueTable
+   */
+  public ColumnVector lowerBound(Table inputTable, Table valueTable, boolean[] descFlags,
+                                 boolean nullsAsLargest) {
+    return new ColumnVector(gdfBound(inputTable.nativeHandle, valueTable.nativeHandle,
+            descFlags, nullsAsLargest, false));
+  }
+
+  /**
+   * Given a sorted inputTable return the upper bound.
+   * Example:
+   *
+   *  Single column:
+   *      idx            0   1   2   3   4
+   *   inputTable  =   { 10, 20, 20, 30, 50 }
+   *   valuesTable =   { 20 }
+   *   result      =   { 3 }
+   *
+   *  Multi Column:
+   *      idx                0    1    2    3    4
+   *   inputTable      = {{  10,  20,  20,  20,  20 },
+   *                      { 5.0,  .5,  .5,  .7,  .7 },
+   *                      {  90,  77,  78,  61,  61 }}
+   *   valuesTable     = {{ 20 },
+   *                      { .7 },
+   *                      { 61 }}
+   *   result          = {  5 }
+   * NaNs in column values do not work especially
+   * if the table has multiple columns.
+   * Strings and String categories do not work for this method. If the input table is
+   * unsorted the results are wrong. Types of columns can be mixed and the result gives the index
+   * of the value pertaining to the matching type column in the input table.
+   * @param inputTable the sorted table as input for bounds
+   * @param valueTable the table of values that need to be inserted
+   * @param descFlags indicates the ordering of the column(s), true if descending
+   * @param nullsAsLargest true if nulls are assumed largest
+   * @return ColumnVector with upper bound indices for all rows in valueTable
+   */
+  public ColumnVector upperBound(Table inputTable, Table valueTable, boolean[] descFlags,
```

Comment:
Same comments here as for lowerBound.  `inputTable` should be `this`, we should follow the orderBy API wherever possible., and we need to make sure that we have some kind of error checking.